client/conf/server.properties.in

```diff
@@ -0,0 +1,41 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# The binding interface used by the management server
+bind.interface=0.0.0.0
```

Comment:
Can we change *0.0.0.0* to *::* so that it listens on IPv6 by default?

I tested this on Ubuntu 16.04 and works fine:

<pre>
root@cloudstack-man-test:~# curl -I http://127.0.0.1:8080/client/
HTTP/1.1 200 OK
Content-Type: text/html
ETag: W/"114557-1499056602000
Content-Length: 114557
Server: Jetty(9.2.z-SNAPSHOT)

root@cloudstack-man-test:~# curl -I http://[::1]:8080/client/
HTTP/1.1 200 OK
Content-Type: text/html
ETag: W/"114557-1499056602000
Content-Length: 114557
Server: Jetty(9.2.z-SNAPSHOT)

root@cloudstack-man-test:~#
</pre>

Reply:
Hi @wido I think the standard I've seen in mysql and elsewhere is to use this. It's a matter of config issue, and ipv6 may not be enabled on all systems.

For me it fails when JAVA_OPT has `-Djava.net.preferIPv4Stack=true` set, and bind.interface has `::` set. Exception seen is:
```
WARN  [o.e.j.u.c.AbstractLifeCycle] (main:null) (logid:) FAILED ServerConnector@4e9ba398{HTTP/1.1}{:::8080}: java.net.SocketException: Protocol family unavailable
java.net.SocketException: Protocol family unavailable
	at sun.nio.ch.Net.bind0(Native Method)
```
Command ran to test this case (build cloudstack and cd to client/target, fix conf/server.properties): `java -Dlog4j.configuration=conf/log4j-cloud.xml -cp cloud-client-ui-4.11.0.0-SNAPSHOT.jar:lib/bcpkix-jdk15on-1.58.jar:lib/bcprov-jdk15on-1.58.jar:.:/usr/share/java/mysql-connector-java.jar:conf:common:utilities/scripts/db/ -Djava.net.preferIPv4Stack=true  org.apache.cloudstack.ServerDaemon`

Reply:
Yes, that's true. However, all modern Linux systems have IPv6 enabled. It's a manual action to disable IPv6 on a system.

You don't need a public IPv6 address though to listen on *::*, you already have your localhost *::1* to listen on.

Don't forget that the *current* Tomcat6/7 already listen on IPv6 by default. So it would be a change from 4.10 to 4.11 if we wouldn't listen on IPv6 anymore.

Reply:
@wido I agree with you, however, for the default I needed to put some configuration that would work out of the box on most/all systems. I know several installations where they have put `-Djava.net.preferIPv4Stack=true ` in their global JAVA opts (or in cloudstack specific env/init/systemd files). That said, we can have a compromise where I can remove the default value set to empty, i.e. `bind.interface=`, that way the system picks up the default interface that could work automatically.

With this approach, on systems without ipv6, this listens on:
tcp        0      0 127.0.0.1:8080          0.0.0.0:*               LISTEN     

And with ipv6 enabled, it listens on:
tcp6       0      0 127.0.0.1:8080          :::*                    LISTEN

Reply:
@wido check now, I've made the above change.